addons/sale/views/sale_views.xml

```diff
@@ -350,7 +350,7 @@
                                     <field name="date_order" attrs="{'invisible': [('state', 'in', ['sale', 'done', 'cancel'])]}"/>
                                     <field name="fiscal_position_id" options="{'no_create': True}"/>
                                     <field name="invoice_status"
-                                        attrs="{'invisible': [('state', 'not in', ('sale','done'))]}"/>
+                                        attrs="{'invisible': [('state', 'not in', ('sale','done'))]}" invisible="1"/>
```

Comment:
invisible [1=1] here
invisible [('state', 'not in', ('sale','done'))] when activating invoicing

Or try with a states=.., like you want.